test/test_mkldnn.py

```diff
@@ -148,6 +148,17 @@ def test_avg_pool2d(self):
                 avg_pool2d(x),
                 avg_pool2d(x.to_mkldnn()).to_dense())
 
+    def test_adaptive_avg_pool2d(self):
+        N = torch.randint(3, 10, (1,)).item()
+        C = torch.randint(3, 10, (1,)).item()
+        x = torch.randn(N, C, 224, 224, dtype=torch.float32) * 100
+
+        adaptive_avg_pool2d = torch.nn.AdaptiveAvgPool2d(7)
+
+        self.assertEqual(
+            adaptive_avg_pool2d(x),
+            adaptive_avg_pool2d(x.to_mkldnn()).to_dense())
```

Comment:
Does it mean that everytime we call mkldnn ops we are required to call `to_dense` ?

Reply:
not every op, but every chunk of ops (in this case the unittest is only testing one op so yeah it's calling to_dense after each op to convert the mkldnn tensor to dense tensor so that we can compare with the regular dense kernel's output).
basically one needs to do:
```
to_mkldnn -> mkldnn_op1 -> mkldnn_op2 ... -> mkldnn_opn -> to_dense
```